File: src/main/java/net/alephdev/calendar/models/Task.java
package net.alephdev.calendar.models;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.ForeignKey;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import java.util.List;
import lombok.Data;

@Data
@Entity
@Table(name = "Task")
public class Task {
  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Integer id;

  @Column(nullable = false)
  private String name;

  private Integer storyPoints;

  @ManyToOne
  @JoinColumn(name = "implementer", foreignKey = @ForeignKey(name = "fk_task_implementer"))
  private User implementer;

  @ManyToOne
  @JoinColumn(name = "sprint_id", foreignKey = @ForeignKey(name = "fk_task_sprint_id"))
  private Sprint sprint;

  @ManyToOne
  @JoinColumn(
      name = "status_id",
      foreignKey = @ForeignKey(name = "fk_task_status_id"),
      nullable = false)
  private Status status;

  @Enumerated(EnumType.STRING)
  @Column(nullable = false)
  private Priority priorityEnum;

  @ManyToOne
  @JoinColumn(name = "created_by", foreignKey = @ForeignKey(name = "fk_task_created_by"))
  private User createdBy;

  @OneToMany(mappedBy = "task")
  @JsonIgnore
  private List<TaskTag> taskTags;

  public enum Priority {
    LOW,
    MEDIUM,
    CRITICAL
  }
}

File: angular/src/app/models/task.ts
import {User} from "./user";
import {Sprint} from "./sprint";
import {Status} from "./status";

/**
 * Модель задачи (Task) в системе управления проектами.
 *
 * Представляет задачу, которая может быть назначена пользователю,
 * привязана к спринту и имеет определенный статус выполнения.
 *
 * @property id - Уникальный идентификатор задачи
 * @property name - Название задачи
 * @property storyPoints - Оценка сложности в Story Points (может быть null)
 * @property implementer - Исполнитель задачи (может быть null если не назначен)
 * @property sprint - Спринт, к которому привязана задача (может быть null)
 * @property status - Текущий статус задачи (To Do, In Progress, Done и т.д.)
 * @property priorityEnum - Приоритет задачи
 * @property createdBy - Пользователь, создавший задачу
 */
export interface Task {
  id: number;
  name: string;
  storyPoints: number | null;
  implementer: User | null;
  sprint: Sprint | null;
  status: Status;
  priorityEnum: TaskPriority;
  createdBy: User;
}

/**
 * Перечисление приоритетов задач.
 *
 * Определяет уровни важности задачи для приоритизации работы команды.
 *
 * @enum {string}
 * @property LOW - Низкий приоритет (не критичные улучшения)
 * @property MEDIUM - Средний приоритет (стандартные задачи)
 * @property CRITICAL - Критический приоритет (требует немедленного внимания)
 */
export enum TaskPriority {
  LOW = 'LOW',
  MEDIUM = 'MEDIUM',
  CRITICAL = 'CRITICAL'
}
